src/cache-utils.ts

```diff
@@ -47,11 +51,24 @@ export const getCommandOutput = async (toolCommand: string) => {
   return stdout.trim();
 };
 
-const getPackageManagerVersion = async (
+export const getPackageManagerWorkingDir = (): string | null => {
+  const cache = core.getInput('cache');
+  if (cache !== 'yarn') {
+    return null;
+  }
```

Comment:
I think we should not do it for yarn specifically. We can remove it.

Reply:
It is too dangerous, unpredictable and was not requested. Yarn requires `cd` to the subdir in order to fix the problem with the incorrect version of yarn, i am not sure how the other package managers will be affected by forcing the to run in the other dir than root `package.json`

If we want to let other project managers to be run in not-root directory we should add `project-dir` input. Setting it explicitly the users accept the possible negative consequences.

Reply:
leave it